get-mumble_deps.sh

```diff
@@ -5,9 +5,26 @@
 # can be found in the LICENSE file in the source tree or at
 # <http://mumble.info/mumble-releng-experimental/LICENSE>.
 
-# Make sure the command-prompt stays open if an error is encountered so that the user can read
-# the error message before the console closes.
-trap "printf '\n\n'; read -p 'ERROR encountered... Press Enter to exit'" ERR
+# Helper function to check if a certain parameter has been passed to the script
+# Taken from https://stackoverflow.com/a/56431189/3907364
+has_param() {
+    local term="$1"
+    shift
+    for arg; do
+        if [[ $arg == "$term" ]]; then
+            return 0
+        fi
+    done
+    return 1
+}
+
+if ! has_param '--auto' "$@"; then
+	# Make sure the command-prompt stays open if an error is encountered so that the user can read
+	# the error message before the console closes.
+	# If you run call this script as part of some automation, you'll want to pass --auto
+	# to make sure you don't get stuck.
+	trap "printf '\n\n'; read -p 'ERROR encountered... Press Enter to exit'" ERR
+fi
```

Comment:
If you could paraphrase (rename) your variable and function names, I don't believe a citation is necessary. These look like normally available logical statements that would be found in the documentation. We can invite another person in the group to review that to confirm.

Reply:
But `has_param` actually describes what it is checking. It doesn't get anything, so `get_param_name` doesn't feel right to me...

Which variable do you want me to rename? `arg`? or `term`?

Reply:
> But `has_param` actually describes what it is checking. It doesn't get anything, so `get_param_name` doesn't feel right to me...
> 
I will meet you in the middle with `has_option` because I think it is technically an option.
> Which variable do you want me to rename? `arg`? or `term`?
`term` should probably be something less ambiguous. I'm also not sure I am reading the function body correctly, but should there be a `for` when only one input is being evaluated? I also feel like `return 1` should probably be in an `else` for the check condition for safety.

Reply:
I think this might need to be considered, also:

https://stackoverflow.com/a/402410

Reply:
`has_option` is fine :)

> I'm also not sure I am reading the function body correctly, but should there be a for when only one input is being evaluated?

Yes. The loop iterates over all input argument and not over all arguments you are searching for. Right now we wouldn't even have to bother with this at all. We could simply check if `$1` is equal to `--auto`. I wanted to not make assumption about the position of the option in the argument list kin case we want to add multiple options at some point.

> I also feel like return 1 should probably be in an else for the check condition for safety.

No that would break the functionality. We only want to return 1 if we're sure that the option hasn't been provided at any place in the argument list.

I went without `getopts` as I wasn't sure if this would work under Windows as well...

Reply:
I will probably end up adding a param for `VCPKGDIR` since people won't necessarily want to have 60+GB of deps on their primary drive if they are running something like SSD+HDD setup.

With regard to the `return` statements, you are returning `0` when something is technically `true` and then returning `1`. Why not just `return 1` when the option is found and break out of the loop? Then the `else` could be `return 0` indicating that it was not found.

Reply:
No it's not returning 0 and then returning 1. If it returns 0 then that's it.
In bash the conversion between bool and int is exactly the other way around as in c(++): `0` means `true` and everything else means `false`. That's because in Bash these numbers are interpreted as exit codes. `0` is good, everything else is bad.

Reply:
Right, because they are exit codes. Thanks for clarifying.

Reply:
No problem :)